### DISCUSSION TITLE
pipeline shouldn't be trigged when there is merge conflict

### DISCUSSION THREAD

**Discussion by y0zg** [jenkinsci/bitbucket-push-and-pull-request-plugin#208] [Ideas]
At least, pipeline shouldn't be triggered when there is merge conflict.
Second point I wonder how to identify properly such merge conflicts with jenkins pipeline like it's done in bitbucket.
Any thoughts?

**Comment by cdelmonte-zg** [comment]
Hi @y0zg 

Thanks, it's a good point.
I would move this issue into the discussions. Is it okay for you?

**Comment by y0zg** [comment]
hi @cdelmonte-zg , sure!

**Comment by y0zg** [comment]
Here are some details and api reference how to get merge conflict status , @cdelmonte-zg do you think this is doable?

https://community.atlassian.com/t5/Bitbucket-questions/API-v2-endpoint-to-get-Pull-Request-conflict-status/qaq-p/927455

https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/pullrequests/%7Bpull_request_id%7D/diffstat

**Reply by cdelmonte-zg** [reply]
hi @y0zg, might you describe a use case scenario in form of a sequence of actions executed by bitbucket and the Jenkins plugin? Just to understand better how to implement this functionality. Thanks!

**Reply by y0zg** [reply]
Sorry, I wouldn't providing any sequence diagram at this moment, but I expect to have a call to bitbucket API for checking merge conflict status whenever PR request is raised. Does it sound reasonable? If no, I could try to provide more details

**Reply by cdelmonte-zg** [reply]
@y0zg it should be enough to start with, tnx
